app/controllers/api/v0/profile_images_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Api
+  module V0
+    class ProfileImagesController < ApiController
+      before_action :set_cache_control_headers, only: %i[show]
+
+      def show
+        not_found unless profile_image_owner
+
+        @profile_image_url = profile_image_owner.profile_image_url
+      end
+
+      private
+
+      def profile_image_owner
+        user || organization
+      end
+
+      def user
+        @user ||= User.select(:id, :profile_image)
+          .find_by(username: params[:username], registered: true)
```

Comment:
love that you thought about the registered users. Can I suggest making it even more explicit by using the `.registered` scope?

Something like

```ruby
User.registered.select().find_by()
```